### PR TITLE
Add --raw-key-events

### LINKED ISSUE #2816: Hope to support the following special ASCII character input, thank you!
The following is the special ASCII character input supported by Qtscrcpy. I hope that scrcpy can also support the following special ASCII character input. I am not good at C language, so come to ask for help, thank you!

```diff
@@ -279,6 +279,7 @@ AndroidKeycode InputConvertNormal::convertKeyCode(int key, Qt::KeyboardModifiers
        keyCode = AKEYCODE_0;
        break;
    case Qt::Key_1:
    case Qt::Key_Exclam:// !
        keyCode = AKEYCODE_1;
        break;
    case Qt::Key_2:
@@ -288,15 +289,19 @@ AndroidKeycode InputConvertNormal::convertKeyCode(int key, Qt::KeyboardModifiers
        keyCode = AKEYCODE_3;
        break;
    case Qt::Key_4:
    case Qt::Key_Dollar://$
        keyCode = AKEYCODE_4;
        break;
    case Qt::Key_5:
    case Qt::Key_Percent:// %
        keyCode = AKEYCODE_5;
        break;
    case Qt::Key_6:
    case Qt::Key_AsciiCircum: //^
        keyCode = AKEYCODE_6;
        break;
    case Qt::Key_7:
    case Qt::Key_Ampersand: //&
        keyCode = AKEYCODE_7;
        break;
    case Qt::Key_8:
@@ -308,6 +313,67 @@ AndroidKeycode InputConvertNormal::convertKeyCode(int key, Qt::KeyboardModifiers
    case Qt::Key_Space:
        keyCode = AKEYCODE_SPACE;
        break;
    case Qt::Key_Comma://,
    case Qt::Key_Less://<
        keyCode = AKEYCODE_COMMA;
        break;
    case Qt::Key_Period://.
    case Qt::Key_Greater://>
        keyCode = AKEYCODE_PERIOD;
        break;
    case Qt::Key_Minus://-
    case Qt::Key_Underscore: //_
        keyCode = AKEYCODE_MINUS;
        break;
    case Qt::Key_Equal://=
        keyCode = AKEYCODE_EQUALS;
        break;
    case Qt::Key_BracketLeft://[
    case Qt::Key_BraceLeft: //{
        keyCode = AKEYCODE_LEFT_BRACKET;
        break;
    case Qt::Key_BracketRight://]
    case Qt::Key_BraceRight: //}
        keyCode = AKEYCODE_RIGHT_BRACKET;
        break;
    case Qt::Key_Backslash:// \ ????
    case Qt::Key_Bar: //|
        keyCode = AKEYCODE_BACKSLASH;
        break;
    case Qt::Key_Semicolon://;
    case Qt::Key_Colon: //:
        keyCode = AKEYCODE_SEMICOLON;
        break;
    case Qt::Key_Apostrophe://'
    case Qt::Key_QuoteDbl: //"
        keyCode = AKEYCODE_APOSTROPHE;
        break;
    case Qt::Key_Slash:// /
    case Qt::Key_Question://?
        keyCode = AKEYCODE_SLASH;
        break;
    case Qt::Key_At://@
        keyCode = AKEYCODE_AT;
        break;
    case Qt::Key_Plus://+
        keyCode = AKEYCODE_PLUS;
        break;
    case Qt::Key_QuoteLeft://`
    case Qt::Key_AsciiTilde://~
        keyCode = AKEYCODE_GRAVE;
        break;
    case Qt::Key_NumberSign: //#
        keyCode = AKEYCODE_POUND;
        break;
    case Qt::Key_ParenLeft: //(
        keyCode = AKEYCODE_NUMPAD_LEFT_PAREN;
        break;
    case Qt::Key_ParenRight: //)
        keyCode = AKEYCODE_NUMPAD_RIGHT_PAREN;
        break;
    case Qt::Key_Asterisk: //*
        keyCode = AKEYCODE_STAR;
        break;
    }
    return keyCode;
}
```

### PR DESCRIPTION
**Pull request by rom1v**
This option allows to inject all input keys as key events, and ignore text events.

Fixes #2816.

---

Here are binaries so that you can test easily. Please replace these files in your v1.20 release:

 - [`scrcpy.exe`](https://tmp.rom1v.com/scrcpy/2827/4/scrcpy.exe) _sha256:c3d96408d81d4c3f17259a8d99bd2e343f6c9758dc85e6317b114f6cdc95fd00_
 - [`scrcpy-server`](https://tmp.rom1v.com/scrcpy/2827/4/scrcpy-server) _sha256:e08e9aa4796c213bf601f6323b5e9f4fce250b1f3b1d46441289824cb4af294a_

<details><summary>old binaries v1</summary>

 - [`scrcpy.exe`](https://tmp.rom1v.com/scrcpy/2831/1/scrcpy.exe) _sha256:848fc28d0b2e864fabcd82f327ddb1ee987fc364f441deed262b8f733fe7298d_
 - [`scrcpy-server`](https://tmp.rom1v.com/scrcpy/2831/1/scrcpy-server) _sha256:e08e9aa4796c213bf601f6323b5e9f4fce250b1f3b1d46441289824cb4af294a_

</details>

This is based on the development branch, and includes #2827, so this is almost a release candidate for scrcpy 1.21.

Please test and report any issue you might encounter. In particular, please test the `--tcpip` option detailed in #2827.

Thank you.

### CODE REVIEW

**Comment by Helaer** [comment]
> This option allows to inject all input keys as key events, and ignore text events.
> 
> Fixes #2816.
> 
> Here are binaries so that you can test easily. Please replace these files in your v1.20 release:
> 
> * [`scrcpy.exe`](https://tmp.rom1v.com/scrcpy/2831/1/scrcpy.exe) _sha256:848fc28d0b2e864fabcd82f327ddb1ee987fc364f441deed262b8f733fe7298d_
> * [`scrcpy-server`](https://tmp.rom1v.com/scrcpy/2831/1/scrcpy-server) _sha256:e08e9aa4796c213bf601f6323b5e9f4fce250b1f3b1d46441289824cb4af294a_
> 
> This is based on the development branch, and includes #2827, so this is almost a release candidate for scrcpy 1.21.
> 
> Please test and report any issue you might encounter. In particular, please test the `--tcpip` option detailed in #2827.
> 
> Thank you.

After my test, after --raw-key-events is enabled, there are the following problems. The left parenthesis cannot be entered, and the left parenthesis cannot be entered in both English and non-English states.

File: app/src/keyboard_inject.c [hunk 1]
```diff
@@ -6,102 +6,192 @@
...
+        {SDLK_0,             AKEYCODE_0},
+        {SDLK_1,             AKEYCODE_1},
+        {SDLK_2,             AKEYCODE_2},
+        {SDLK_3,             AKEYCODE_3},
+        {SDLK_4,             AKEYCODE_4},
+        {SDLK_5,             AKEYCODE_5},
+        {SDLK_6,             AKEYCODE_6},
+        {SDLK_7,             AKEYCODE_7},
+        {SDLK_8,             AKEYCODE_8},
+        {SDLK_8,             AKEYCODE_9},
```
Comment (rom1v) [inline app/src/keyboard_inject.c:125]: Oops.

**Comment by rom1v** [comment]
Thank you. Indeed, there is a typo for the 9 :+1:

**Comment by rom1v** [comment]
I pushed a new version, please re-test :wink:

**Comment by Helaer** [comment]
> 我推送了一个新版本，请重新测试 😉

--raw-key-events works well on my device, you can merge it.

**Pull request merged**